index.js

```diff
@@ -37,5 +46,16 @@ module.exports = (babel, userOptions) => {
     [require('@babel/preset-react').default, {useBuiltIns}],
   ]
 
+  if (typescript) {
+    // eslint-disable-next-line import/no-extraneous-dependencies
+    const typescriptModule = require('@babel/preset-typescript').default
+
+    if (typescript instanceof Object) {
+      presets.push([typescriptModule, typescript])
+    } else {
+      presets.push(typescriptModule)
+    }
+  }
```

Comment:
It doesn't make sense to include both the Flow and Typescript presets, so can we consolidate these into a single option to choose whether to support Flow, Typescript, or neither?

Reply:
I've updated it to simply exclude the Flow preset if the `typescript` options is present. But do you think a `typechecker`: `flow` | `typescript` type of argument would be better?